apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/ProfilingActivationListener.java

```diff
@@ -46,7 +47,7 @@ public ProfilingActivationListener(ElasticApmTracer tracer) {
 
     @Override
     public void beforeActivate(TraceContextHolder<?> context) throws Throwable {
-        if (!context.isSampled()) {
+        if (!context.isSampled() || context instanceof ErrorCapture) {
```

Comment:
Do the ` || context instanceof ErrorCapture` check on `afterDeactivate` as well.

Reply:
done